packages/myst-parser/src/directives.ts

```diff
@@ -208,7 +208,7 @@ export function applyDirectives(
     node.children = run(data, vfile, {
       // Implement a parseMyst function that accepts _relative_ line numbers
       parseMyst: (source: string, offset: number = 0) =>
-        ctx.parseMyst(source, offset + node.pos[0]),
```

Comment:
Oof, looks like I took this verbatim without testing it. I hadn't actually gotten to validation yet.

Reply:
That was a hard error. :)

The testing was helpful to pull out the other issue of recursive parsing, which I did not expect at all!

I also don't think this would have been easy to see in the next two PRs, so glad we have some test coverage on this at the parser level!